torch/_meta_registrations.py

```diff
@@ -6510,6 +6510,30 @@ def meta__padded_dense_to_jagged_forward(
     return padded.new_empty(output_shape)
 
 
+@register_meta(aten.split_with_sizes)
+def meta_split_with_sizes(
+    self: Tensor, split_sizes: List[int], dim: int = 0
+) -> List[Tensor]:
+    # This method has a decomposition, so this function is not necessary
+    # but running the decomposition with meta tensors is slow
+    for split_size in split_sizes:
+        torch._check_is_size(
+            split_size,
+            lambda: "split_with_sizes expects split_sizes have only non-negative entries",
+        )
+    torch._check_with(
+        ValueError,
+        sum(split_sizes) == self.shape[dim],
+        lambda: f"Split sizes add up to {sum(split_sizes)} but got the tensor's size of {self.shape[dim]}",
+    )
+    result = []
+    for split_size in split_sizes:
+        shape = list(self.shape)
+        shape[dim] = split_size
+        result.append(self.new_empty(shape))
```

Comment:
oh no, you can't do this lol, you have to actually return views 😭

Reply:
Ah, that makes the difference.